src/PKSim.Core/Snapshots/Mappers/ParameterContainerSnapshotMapperBase.cs

```diff
@@ -46,17 +48,20 @@ protected virtual bool ParameterHasChanged(IParameter parameter)
          return parameter.ValueDiffersFromDefault() || canBeEdited && parameter.Value != 0;
       }
 
-      protected void UpdateParametersFromSnapshot(TSnapshot snapshot, IContainer container, string containerDesciptor)
+      protected Task UpdateParametersFromSnapshot(TSnapshot snapshot, IContainer container, string containerDesciptor)
       {
+         var tasks = new List<Task>();
          foreach (var snapshotParameter in snapshot.Parameters)
          {
             var modelParameter = container.Parameter(snapshotParameter.Name);
 
             if (modelParameter == null)
-               throw new SnapshotParameterNotFoundException(snapshotParameter.Name, containerDesciptor);
+               return FromException(new SnapshotParameterNotFoundException(snapshotParameter.Name, containerDesciptor));
```

Comment:
This methos (Task.FromException) is defined in ,NET 4.6. I had to create my own for now until we upgrade